drivers/spi/spi_nrf5.c

```diff
@@ -0,0 +1,481 @@
+			return NRFX_SPI_MODE_1;
+		} else {
+			return NRFX_SPI_MODE_0;
+		}
+	}
+}
+
+static inline
+nrfx_spi_bit_order_t spi_operation_to_nrfx_bit_order(u16_t operation)
+{
+	if (operation & SPI_TRANSFER_LSB) {
+		return NRFX_SPI_BIT_ORDER_LSB_FIRST;
+	} else {
+		return NRFX_SPI_BIT_ORDER_MSB_FIRST;
+	}
+}
+
+static inline int set_nrfx_spi_config(nrfx_spi_config_t *nrfx_spi_cfg,
+				      const struct spi_config *spi_cfg)
+{
+	const struct spi_nrf5_cfg *dev_cfg = DEV_CFG(spi_cfg);
+	u8_t slave = spi_cfg->slave;
+
+	nrfx_spi_cfg->sck_pin  = dev_cfg->psel.sck;
+	nrfx_spi_cfg->mosi_pin = dev_cfg->psel.mosi;
+	nrfx_spi_cfg->miso_pin = dev_cfg->psel.miso;
+
+	if (spi_cfg->cs) {
+		nrfx_spi_cfg->ss_pin = NRFX_SPI_PIN_NOT_USED;
+	} else {
+		if (slave < SPI_NRF5_MAX_SLAVES_ON_BUS) {
+			if (dev_cfg->psel.ss[slave] ==
+			    CONFIG_SPI_NRF5_SS_UNUSED_PIN) {
+				return -1;
+			}
+
+			nrfx_spi_cfg->ss_pin = dev_cfg->psel.ss[slave];
+		} else {
+			return -1;
+		}
+	}
+
+	/* IRQ priority is set via IRQ_CONNECT */
+	nrfx_spi_cfg->irq_priority = 0;
+	nrfx_spi_cfg->orc = dev_cfg->orc;
+	nrfx_spi_cfg->frequency = spi_freq_to_nrfx_freq(spi_cfg->frequency);
+	nrfx_spi_cfg->mode = spi_operation_to_nrfx_mode(spi_cfg->operation);
+	nrfx_spi_cfg->bit_order = spi_operation_to_nrfx_bit_order(
+							spi_cfg->operation);
+
+	return 0;
+}
+
+static void spi_nrf5_nrfx_evt_handler(const nrfx_spi_evt_t *p_event,
+				      void *p_context);
+
+static int spi_nrf5_configure(const struct spi_config *spi_cfg)
+{
+	const struct spi_nrf5_cfg *dev_cfg = DEV_CFG(spi_cfg);
+	struct spi_nrf5_data *dev_data = DEV_DATA(spi_cfg);
+
+	SYS_LOG_DBG("%p (prev %p)", spi_cfg, dev_data->ctx.config);
+
+	if (spi_context_configured(&dev_data->ctx, spi_cfg)) {
+		/* Nothing to do */
+		return 0;
+	}
+
+	if (spi_cfg->operation & SPI_OP_MODE_SLAVE) {
+		SYS_LOG_ERR("SPI peripheral does not support slave mode");
+		return -EINVAL;
+	}
+
+	if ((spi_cfg->operation & SPI_LINES_MASK) != SPI_LINES_SINGLE) {
+		SYS_LOG_ERR("SPI peripheral does not support dual, quad, "
+			    "or octal mode");
+		return -EINVAL;
+	}
+
+	if (SPI_WORD_SIZE_GET(spi_cfg->operation) != 8) {
+		SYS_LOG_ERR("Nordic SPI does not support word size other "
+			    "than 8 bits");
+		return -EINVAL;
+	}
+
+	nrfx_err_t err;
+	nrfx_spi_config_t nrfx_spi_cfg;
+
+	if (dev_data->ctx.config) {
+		nrfx_spi_uninit(&dev_cfg->spi);
+		dev_data->ctx.config = NULL;
+	}
+
+	if (set_nrfx_spi_config(&nrfx_spi_cfg, spi_cfg)) {
+		return -EINVAL;
+	}
+
+	err = nrfx_spi_init(&dev_cfg->spi,
+			    &nrfx_spi_cfg,
+			    spi_nrf5_nrfx_evt_handler,
+			    spi_cfg->dev);
+
+	if (err != NRFX_SUCCESS) {
+		SYS_LOG_ERR("Error %d while initializing nrfx SPI", err);
+		return -1;
+	}
+
+	dev_data->ctx.config = spi_cfg;
+
+	spi_context_cs_configure(&dev_data->ctx);
+
+	return 0;
+}
+
+static inline void set_buffers_len_for_nrfx(nrfx_spi_xfer_desc_t *xfer_desc,
+					    u32_t len)
+{
+	if (xfer_desc->p_tx_buffer) {
```

Comment:
Is it somewhere taken in account that the length can not be larger then 255 for nrf58240 aa?